### PR TITLE
Add JUnit Platform JFR in Dependency Metadata and Diagram

### LINKED ISSUE #2403: Include junit-platform-jfr module artifact in Appendix in User Guide
## Overview

The `junit-platform-jfr` artifact introduced in JUnit Platform 1.7.0 has not yet been added to the [Dependency Metadata](https://junit.org/junit5/docs/current/user-guide/#dependency-metadata) and [Dependency Diagram](https://junit.org/junit5/docs/current/user-guide/#dependency-diagram) sections of the User Guide.

## Deliverables

- [ ] Add `junit-platform-jfr` artifact to the `Dependency Metadata` section of the User Guide.
- [ ] Add `junit-platform-jfr` artifact to the `Dependency Diagram` section of the User Guide.

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

This commit also removes the dependency metadata from the `README.md` file and replaces it with a link to the user guide instead.

Closes #2403

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: documentation/src/docs/asciidoc/user-guide/appendix.adoc [hunk 1]
```diff
@@ -39,6 +39,8 @@ artifacts are deployed to Sonatype's {snapshot-repo}[snapshots repository] under
     directory. See <<running-tests-console-launcher>> for details.
   `junit-platform-engine`::
     Public API for test engines. See <<launcher-api-engines-custom>> for details.
+  `junit-platform-jfr`::
+    Provides the JUnit Platform Flight Recording Listener. See <<running-tests-flight-recorder>> for details.
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/user-guide/appendix.adoc:43]: How about something like "Provides a Java Flight Recorder `TestExecutionListener` for the JUnit Platform" or "Provides a `TestExecutionListener` for Java Flight Recorder events on the JUnit Platform"?

**Review by marcphilipp — APPROVED** [review]

File: documentation/src/docs/asciidoc/user-guide/appendix.adoc (hunk 1 above)
Comment (sormuras) [reply documentation/src/docs/asciidoc/user-guide/appendix.adoc:43]: I take the second sentence.
